collect_app/src/main/java/org/odk/collect/android/forms/DatabaseFormRepository.java

```diff
@@ -48,4 +49,32 @@ public void save(Form form) {
     public void delete(Long id) {
         new FormsDao().deleteFormsFromIDs(new String[]{id.toString()});
     }
+
+    @Override
+    public void setLastDetectedUpdated(String jrFormId, String formHash, String manifestHash) {
+        String formVersionHash = MultiFormDownloader.getMd5Hash(formHash) + manifestHash;
+
+        ContentValues values = new ContentValues();
+        values.put(LAST_DETECTED_FORM_VERSION_HASH, formVersionHash);
+        new FormsDao().updateForm(values, JR_FORM_ID + "=?", new String[]{jrFormId});
+    }
+
+    @Override
+    public Form getByLastDetectedUpdate(String formHash, String manifestHash) {
```

Comment:
I was actually trying to abstract away from the "hash" here. In the cases we're using it we want to know if this "update" or "version" of the form exists on the device. We don't need to hash anything to do that (it could have just stored the manifest hash rather than hashing it with the form hash) so to me this felt closer to what the intention was while hiding the implementation detail. Does that feel ok?

Reply:
Fair enough, thanks for the explanation.